crates/types/src/block_impl.rs

```diff
@@ -153,29 +155,38 @@ impl BlockPayload for VIDBlockPayload {
         )
     }
 
-    fn encode(&self) -> Vec<u8> {
-        VIDTransaction::encode(self.transactions.clone())
-    }
-
-    fn decode(encoded_transactions: &[u8]) -> Self {
+    fn from_bytes(encoded_transactions: Self::Encode<'_>, _metadata: Self::Metadata) -> Self {
+        let encoded_vec: Vec<u8> = encoded_transactions.collect();
         let mut transactions = Vec::new();
         let mut current_index = 0;
-        while current_index < encoded_transactions.len() {
-            let txn_len = encoded_transactions[current_index] as usize;
-            let next_index = current_index + txn_len;
+        while current_index < encoded_vec.len() {
+            // Decode the length of the transaction and the transaction bytes.
+            let txn_start_index = current_index + size_of::<usize>();
```

Comment:
Same here, we should replace `usize` with something platform-independent, e.g. `u32`